game-core/src/main/java/games/strategy/engine/framework/startup/ui/editors/ForumPosterEditor.java

```diff
@@ -6,179 +6,181 @@
 import java.awt.image.BufferedImage;
 import java.io.File;
 import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Level;
 
 import javax.imageio.ImageIO;
 import javax.swing.JButton;
 import javax.swing.JCheckBox;
+import javax.swing.JComboBox;
 import javax.swing.JLabel;
 import javax.swing.JOptionPane;
-import javax.swing.JPasswordField;
 import javax.swing.JTextField;
 import javax.swing.SwingUtilities;
-import javax.swing.event.DocumentListener;
 
 import com.google.common.base.Preconditions;
 
 import games.strategy.engine.ClientContext;
+import games.strategy.engine.data.properties.GameProperties;
 import games.strategy.engine.framework.GameRunner;
+import games.strategy.engine.pbem.AxisAndAlliesForumPoster;
 import games.strategy.engine.pbem.IForumPoster;
+import games.strategy.engine.pbem.TripleAForumPoster;
+import games.strategy.triplea.settings.ClientSetting;
 import games.strategy.ui.ProgressWindow;
 import games.strategy.util.TimeManager;
+import games.strategy.util.Util;
+import lombok.extern.java.Log;
 
 /**
  * A class for selecting which Forum poster to use.
  */
+@Log
 public class ForumPosterEditor extends EditorPanel {
   private static final long serialVersionUID = -6069315084412575053L;
   private final JButton viewPosts = new JButton("View Forum");
   private final JButton testForum = new JButton("Test Post");
-  private final JLabel loginLabel = new JLabel("Login:");
-  private final JLabel passwordLabel = new JLabel("Password:");
-  private final JTextField login = new JTextField();
-  private final JTextField password = new JPasswordField();
   private final JTextField topicIdField = new JTextField();
   private final JLabel topicIdLabel = new JLabel("Topic Id:");
+  private final JLabel forumLabel = new JLabel("Forums:");
   private final JCheckBox includeSaveGame = new JCheckBox("Attach save game to summary");
   private final JCheckBox alsoPostAfterCombatMove = new JCheckBox("Also Post After Combat Move");
-  private final JCheckBox credentialsSaved = new JCheckBox("Remember me");
```

Comment:
Yeah, as noted in the PR description I haven't really figured out a good way to keep the credentials sort of session temporary. Previously the credentials were simply serialized into the game properties, and simply removed before posting, probably not the best way to do this, simply because it's super easy to forget clearing out the credentials at some point in the code.

If you can point out a good mechanism to achieve this behaviour, I'd be happy to re-introduce this feature in a follow-up PR

Reply:
Well, for this specific issue, it would require restoring all the UI controls for both PBEM/PBF that permit a user to provide credentials that haven't been persisted anywhere.  That covers the case where a user wishes not to store their credentials persistently.

To allow for persistence, maybe we need to look at an approach similar to what you did for PBEM "presets" in the engine preferences, but, in this case, it corresponds to "credential presets."  That is, in the engine preferences, instead of storing a single PBEM configuration and two PBF configurations, maybe we allow for an arbitrary "map" for each?

For example, I could choose to add two Gmail PBEM configurations, one Hotmail PBEM configuration, two `triplea-game.org` PBF configurations, and one `axisandallies.org` PBF configuration.  Then, when I go to the PBEM/PBF setup screen, there is a dropdown button (or something) that allows me to select one of the presets, and that simply fills in the re-added UI controls.

To automatically restore those presets when loading a saved game that has been "passed back" to the user, you could use something similar to what you're already doing.  At least as I understand it, you're storing something in the save game that allows you to cross-reference which PBEM/PBF configuration to use from the engine preferences.  I'm sure that could be extended to support the above proposal where there is an arbitrary number of PBEM/PBF configurations using a heuristic or something.  Worst case, if the heuristic chooses incorrectly, the user can simply use the dropdown button once again to select their preset.

Lots of hand-waving up there, but it's probably doable.  We just need to keep it in mind going forward.  If at all possible, users should always have the ability to opt-out of having us store their sensitive information, no matter how much care _we_ believe we're using.